### DISCUSSION TITLE
画像を指で動かす際の考え方について

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#74] [Question-original]
１　配列で宣言されている画像を指で動かすコードの修正をしたいです。

２　もう一度、昨日よんぱちさんがzoomで教えて頂いたやり方と似たやり方なのですが、自分なりにコードの意味を文字にしてみたので、質問ディスカッションに写真が送れないので個人チャンネルに送ります。
さらに
var viewFrame: CGRect = StoneImage[i].frame
のところにも
Thread 1: Fatal error: Index out of range
同じエラーが出てしまいました。

昨日教えて頂いたやり方と自分がどこで解釈を間違えているかも教えて頂きたいです。

３　https://i-app-tec.com/ios/image-drag.html

４　override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?){
        // タッチイベントを取得
        let touchEvent = touches.first!
        // ドラッグ前の座標, Swift 1.2 から
        let preStonex = touchEvent.previousLocation(in: self.view).x
        let preStoney = touchEvent.previousLocation(in: self.view).y
        // ドラッグ後の座標
        let newStonex = touchEvent.location(in: self.view).x
        let newStoney = touchEvent.location(in: self.view).y
        // ドラッグしたx座標の移動距離
        let Stonedx = newStonex - preStonex
        // ドラッグしたy座標の移動距離
        let Stonedy = newStoney - preStoney
        
        for i in 0...stonenum{
            // 画像のフレーム
            var viewFrame: CGRect = StoneImage[i].frame
            if Stonedx>0{
                // 移動分を反映させる
                viewFrame.origin.x += Stonedx
                viewFrame.origin.y += Stonedy
                StoneImage[i].frame = viewFrame
                self.view.addSubview(StoneImage[i])
            }
            
        }
    }

５　https://github.com/ShunHigu/OriginalNew

**Comment by Yonpachi48** [comment]
何を聞きたいのか明確に記してほしいです！
`var viewFrame: CGRect = StoneImage[i].frame`における`Thread 1: Fatal error: Index out of range`エラーのこと？
それとも画像を動かすコードのこと？？
質問は基本的に一つずつ答えていくので、複数質問がある場合はそれぞれ分けて質問していただきたい！

**Reply by ShunHigu** [reply]
画像を動かすコードについて教えて頂きたいです。
自分の考え方でやったら、このようなエラーが出てしまったということも報告した方が良いと思い報告させて頂きました！
